### DISCUSSION TITLE
how to enforce browser compatibility?

### DISCUSSION THREAD

**Discussion by daniellacosse** [vanilla-extract-css/vanilla-extract#1212] [Ideas]
I'm in the process of migrating a project to vanilla-extract - really liking it - but I've realized I'm losing a lot of what I'd set up via stylelint. In particular, the blocking of CSS properties that aren't supported in my target browsers.

I think I'd get the same thing with some good e2e testing so it's not gonna block me, but I think it would be really cool if vanilla-extract enforced certain things at the type-level based on configs [like browserslist](https://browsersl.ist)

I may try to rig something temporary with [CSS.supports](https://developer.mozilla.org/en-US/docs/Web/API/CSS/supports_static) so stay tuned

**Comment by askoufis** [comment]
It's definitely good practice to use `@supports` queries for fallback behaviour.

Enforcing type-level constraints based on browser targets could be interesting. However, that would likely involve code-generation on our end, which is complexity that could potentially be avoided if we instead created a postCSS custom syntax that enabled [stylint](https://stylelint.io/developer-guide/syntaxes/) to work with VE. That would also be a non-trivial amount of effort though (unless there's an easier way).
